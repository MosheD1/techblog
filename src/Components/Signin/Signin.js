import React, {Component} from 'react'
import './Signin.css'

class Signin extends Component {
    constructor(props) {
        super(props);
        this.state = {
            signinEmail: '',
            signinPassword: ''
        }
    }

    onEmailChange = (e) => {
        this.setState({signinEmail: e.target.value});
    }

    onPasswordChange = (e) => {
        this.setState({signinPassword: e.target.value});
    }

    onSubmitSignIn = () => {
        this.props.onRouteChange('home');
        this.props.loadUser(this.state);
    }

    render() {
        return(
            <main>
                <div className='container signin'>
                    <fieldset>
                        <legend>SignIn</legend>
                        <div className='widget'>
                            <label>Email</label>
                            <input onChange={this.onEmailChange} type='text'></input>
                        </div>
                        <div className='widget'>
                            <label>Password</label>
                            <input onChange={this.onPasswordChange} type='text'></input>
                        </div>
                    </fieldset>
                    <input onClick={() => this.onSubmitSignIn()} type='submit' value='Sign in'></input>
                </div>
        </main>
        );
    }
}

export default Signin;