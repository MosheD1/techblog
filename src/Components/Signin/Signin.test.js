import React from 'react';
import Signin from './Signin';
import {shallow, render, mount} from 'enzyme';

let mockState = {
    signinEmail: '',
    signinPassword: ''
}

describe("Signin Component", () => {
    it('expect to render Signin', () => {
        expect(shallow(<Signin />)).toMatchSnapshot();
    });

    it('expect to get initial state', () => {
        expect(shallow(<Signin />).state()).toEqual(mockState);
    });
});