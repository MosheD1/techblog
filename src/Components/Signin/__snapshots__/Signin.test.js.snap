// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin Component expect to render Signin 1`] = `ShallowWrapper {}`;
